docs/invoke.mdx

```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
+
+```ts
+import { createMachine, createActor, fromPromise, assign } from 'xstate'
+
+const fetchUser = (userId: string) =>
+  fetch(`https://example.com/${userId}`).then((response) => response.text());
+
+const userMachine = createMachine({
+  types: {} as {
+    context: {
+      userId: string,
+      user: object | undefined,
+      error: unknown
+    }
+  },
+  id: 'user',
+  initial: 'idle',
+  context: {
+    userId: '42',
+    user: undefined,
+    error: undefined
+  },
+  states: {
+    idle: {
+      on: {
+        FETCH: { target: 'loading' }
+      }
+    },
+    loading: {
+      // highlight-start
+      invoke: {
+        id: 'getUser',
+        src: fromPromise(({ input }) => fetchUser(input.userId)),
+        input: ({ context: { userId } }) => ({ userId }),
+        onDone: {
+          target: 'success',
+          actions: assign({ user: ({ event }) => event.output })
+        },
+        onError: {
+          target: 'failure',
+          actions: assign({ error: ({ event }) => event.data })
+        }
+      }
+      // highlight-end
+    },
+    success: {},
+    failure: {
+      on: {
+        RETRY: { target: 'loading' }
+      }
+    }
+  }
+});
+```
+
+Actors can also be invoked on the _root_ of the machine, where they will last the lifetime of the machine:
+
+```ts
+import { fromEventObservable, fromEvent } from 'rxjs'
+const interactiveMachine = createMachine({
+  // highlight-start
+  invoke: {
+    src: fromEventObservable(
+      () => fromEvent(document.body, 'click') as Subscribable<EventObject>
+    )
+  },
+  // highlight-end
+  on: {
+    click: {
+      actions: ({ event }) => console.log(event)
+    }
+  }
+});
+```
+
+And `invoke` can be an array, to invoke multiple actors:
+
+```ts
+const vitalsWorkflow = createMachine({
+  states: {
+    CheckVitals: {
+      // highlight-start
+      invoke: [
+        { src: 'checkTirePressure' },
+        { src: 'checkOilPressure' },
+        { src: 'checkCoolantLevel' },
+        { src: 'checkBattery' }
+      ]
+      // highlight-end
+    }
+  }
+})
+
+See [Multiple Actors](#multiple-actors) for more.
+```
+
+For further examples, see:
+- [Reusing function and event definitions workflow](https://github.com/statelyai/xstate/tree/next/examples/workflow-reusing-functions/main.ts)
+- [Check inbox periodically (cron-based workflow)](https://github.com/statelyai/xstate/tree/next/examples/workflow-check-inbox/main.ts)
+- [Car vitals checks (SubFlow Repeat) workflow](https://github.com/statelyai/xstate/tree/next/examples/workflow-car-vitals)
+
+
+## How are Actors Different From Actions?
+
+Actions are “fire-and-forget”; as soon as their execution starts, the statechart running the actions forgets about them. If you specify an action as `async`, **the action won’t be awaited before moving to the next state**.
```

Comment:
```suggestion
Actions are “fire-and-forget”; as soon as their execution starts, the state machine running the actions forgets about them. If you specify an action as `async`, **the action won’t be awaited before moving to the next state**.
```

Reply:
Can we define _zero-time_ here? Might not be a familiar term to beginners? Maybe something like: 

> Remember: transitions are always _zero-time_ (states transition synchronously).

Reply:
Yes, good idea